### PR TITLE
Added effect SinusRunner2.

### PR DESCRIPTION
**Pull request by polovik**
Effect SinusRunner2 - every color is changed by SIN. But phase between sinusoids is adjustable.

### CODE REVIEW

File: src/effect_sinus_runner2.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    //analogWrite(LED_G, 0);
+    analogWrite(LED_R, val_r);
+    analogWrite(LED_G, val_g);
+    analogWrite(LED_B, val_b);
+
+    // Sleep
+    delay(1);
+
+    time += speed;
+    if(time >= 360.0f) {
```
Comment (dns2utf8) [inline src/effect_sinus_runner2.cpp:71]: Missing space  after if

File: src/src.ino [hunk 2]
```diff
@@ -41,30 +42,62 @@ enum Effect {
...
+    int state = digitalRead(BUTTON_MODE);
+    if (state != buttonStatePrev) {
+        if (state == LOW) {
+            effect = (Effect)(effect + 1);
+            if (effect == EffectNone) {
+                effect = RGBControl;
+            }
+        }
+        buttonStatePrev = state;
+    }
```
Comment (dns2utf8) [inline src/src.ino:99]: Would you mind moving this to a function?

**Review by dns2utf8 — CHANGES_REQUESTED** [review]
Thank you for your PR. It looks really good 👍 
Unfortunatly I am currently unable to test it right now. 

What kind of additional hardware did you use?

**Comment by polovik** [comment]
It's strange - I've added only one commit to pull request (941ed7c - "Added effect SinusRunner2 - every color is changed by SIN. But phase between sinusoids is adjustable.") but the second is added too.
I wasn't intent to include the second commit to pull request, because it requeres additional hardware - 1button (for switch effects), 1analog voltage divider (for measure voltage on an accumulator), 1led "L" (for display low charge).

File: src/effect_sinus_runner2.cpp (hunk 1 above)
Comment (polovik) [reply src/effect_sinus_runner2.cpp:71]: I've copied most source from your code - effect_blue_sin.cpp Therefore somewhere "coding style" was remained :)

File: src/src.ino (hunk 2 above)
Comment (polovik) [reply src/src.ino:99]: Please, look at a comment below.

**Comment by dbrgn** [comment]
No problem, we can cherry-pick your first commit :) Thanks for the contribution!

**Comment by dbrgn** [comment]
Merged in a43afcd.

**Pull request closed without merging**